api/grist.yml

```
@@ -666,6 +666,64 @@ paths:
       responses:
         200:
           description: Success
+    put:
+      tags:
+        - /records
+      summary: "Add or update records of a table"
+      parameters:
+        - $ref: '#/components/parameters/docIdPathParam'
+        - $ref: '#/components/parameters/tableIdPathParam'
+        - in: query
+          name: onmany
+          schema:
+            type: string
+            enum:
+              - first
+              - none
+              - all
+            description: >
+              When updating, which matches to update:
+                * `first` - at most one match
+                * `none` - do not update anything
+                * `all` - update all matches
+        - in: query
+          name: noadd
+          schema:
+            type: boolean
+            description: "Set to true to prohibit adding records."
+        - in: query
+          name: noupdate
+          schema:
+            type: boolean
+            description: "Set to true to prohibit updating records."
+        - in: query
+          name: noparse
+          schema:
+            type: boolean
+            description: "Set to true to prohibit converting types."
```

```suggestion
            description: "Set to true to avoid parsing strings according to the column type."
```

Several endpoints actually support this, but it's not mentioned in the others.

---

Updated text (thanks) and documented noparse for /records POST+PATCH and /data POST+PATCH